src/queues/mod.rs

```diff
@@ -98,7 +98,7 @@ impl Queues {
             .receive()
             .and_then(move |messages| {
                 let mut futures: Vec<Box<Future<Item = (), Error = QueueError>>> = Vec::new();
-                for message in messages.into_iter() {
+                for message in messages {
```

Comment:
No need to change it, but this one I have mixed feelings about too fwiw. I quite like explicit use of `into_iter()` as a clear signal that the code is consuming `messages` and it can't be used again after the loop.